### PR TITLE
 Jsonized time to atack skill data

### PR DESCRIPTION
**Pull request by ZhilkinSerg**
#### Summary

```SUMMARY: Infrastructure "Jsonized time to atack skill data "```

#### Purpose of change

Allow defining of time to attack data in json for vanilla and mods skills.

#### Describe the solution

```JSON
  {
    "type": "skill",
    "ident": "smg",
...
    "time_to_attack": { "min_time": 20, "base_time": 80, "time_reduction_per_level": 10 },
...
  },
```

#### Additional context

#35684 can be closed after current PR is merged and mod author should add a new skill in their mod, e.g.:

```JSON
  {
    "type": "skill",
    "ident": "rifle_bolt_action",
    "name": "rifles (bolt-action)",
    "description": "Bolt action is a type of firearm action where the handling of cartridges into and out of the weapon's barrel chamber is operated by manually manipulating the bolt directly via a handle.",
    "tags": [ "combat_skill" ],
    "time_to_attack": { "min_time": 60, "base_time": 600, "time_reduction_per_level": 20 },
    "display_category": "display_ranged"
  },
```

### CODE REVIEW

File: src/skill.h [hunk 1]
```diff
@@ -27,6 +36,7 @@ class Skill
         translation _name;
         translation _description;
         std::set<std::string> _tags;
+        time_info_t _time_to_attack;
```
Comment (ZhilkinSerg) [inline src/skill.h]:
```suggestion
```

File: src/skill.h [hunk 2]
```diff
@@ -27,6 +36,7 @@ class Skill
...
         skill_displayType_id _display_type;
```
Comment (ZhilkinSerg) [inline src/skill.h]:
```suggestion
        skill_displayType_id _display_type;
        time_info_t _time_to_attack;
```

File: src/skill.h [hunk 3]
```diff
@@ -27,7 +36,8 @@ class Skill
         translation _name;
         translation _description;
         std::set<std::string> _tags;
-        skill_displayType_id _display_type;
+         skill_displayType_id _display_type;
```
Comment (ZhilkinSerg) [inline src/skill.h]:
```suggestion
        skill_displayType_id _display_type;
```

File: src/skill.h [hunk 4]
```diff
@@ -27,7 +36,8 @@ class Skill
...
+         time_info_t _time_to_attack;
```
Comment (ZhilkinSerg) [inline src/skill.h]:
```suggestion
        time_info_t _time_to_attack;
```

**Comment by I-am-Erk** [comment]
Loving this push to jsonize skill information

**Comment by kevingranade** [comment]
This pull request has been mentioned on **Cataclysm: Dark Days Ahead**. There might be relevant details there:

https://discourse.cataclysmdda.org/t/new-gun-classes/22972/2

**Pull request merged**